spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/dsl/JdbcStoredProcOutboundGatewaySpec.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.dsl;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.springframework.integration.dsl.ComponentsRegistration;
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.jdbc.StoredProcExecutor;
+import org.springframework.integration.jdbc.StoredProcOutboundGateway;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link MessageHandlerSpec} for a {@link JdbcStoredProcOutboundGatewaySpec}.
+ *
+ * @author Jiandong Ma
+ *
+ * @since 7.0
+ */
+public class JdbcStoredProcOutboundGatewaySpec
+		extends MessageHandlerSpec<JdbcStoredProcOutboundGatewaySpec, StoredProcOutboundGateway>
+		implements ComponentsRegistration {
+
+	private final StoredProcExecutor storedProcExecutor;
+
+	private final StoredProcExecutorConfigurer storedProcExecutorConfigurer;
+
+	protected JdbcStoredProcOutboundGatewaySpec(StoredProcExecutor storedProcExecutor) {
+		this.storedProcExecutor = storedProcExecutor;
+		this.storedProcExecutorConfigurer = new StoredProcExecutorConfigurer(this.storedProcExecutor);
+		this.target = new StoredProcOutboundGateway(this.storedProcExecutor);
+	}
+
+	/**
+	 * Configure the storedProcExecutor through storedProcExecutorConfigurer by invoking the {@link Consumer} callback
+	 * @param configurer the configurer.
+	 * @return the spec
+	 */
+	public JdbcStoredProcOutboundGatewaySpec configurerStoredProcExecutor(Consumer<StoredProcExecutorConfigurer> configurer) {
```

Comment:
Interesting idea, but this is not the style we follow in the project already.
I mean it feels like for the existing and familiar DSL API , it would be natural to have this as a dedicated `StoredProcExecutorSpec` and provide its factory method in that `Jdbc` main entry point.
But we can think about this callback as well.
I mean leave this `configurerStoredProcExecutor` method but have it like this instead:
```
configurerStoredProcExecutor(Consumer<StoredProcExecutorSpec> configurer)
```
So, then end-user would have a choice for their preferences: Use `Jdbc.storedProcExecutor()` factory or this configurer.